django/db/models/expressions.py

```diff
@@ -774,6 +774,7 @@ def set_source_expressions(self, exprs):
 
     def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
         c = self.copy()
+        c.cases = c.cases[:]
```

Comment:
Looking at other expressions, it like this belongs in the `copy()` method? Any reason not to do that?